jetty-http2/http2-common/src/main/java/org/eclipse/jetty/http2/parser/RateControl.java

```diff
@@ -36,4 +36,18 @@
      * @return true IFF the rate is within limits
      */
     public boolean onEvent(Object event);
+
+    /**
+     * Factory to create RateControl instances.
+     */
+    public interface Factory
+    {
+        /**
+         * @return a new RateControl instance.
+         */
+        public default RateControl newRateControl()
```

Comment:
If we are going with a factory pattern, then you may as well pass in the endpoint, so a factory can make different rate controls for different endpoints if need be.

Reply:
Now passing the `EndPoint`.
Hopefully users won't use it to write bytes to the network.